scripts/lib/CIME/XML/env_batch.py

```diff
@@ -273,6 +274,18 @@ def set_job_defaults(self, batch_jobs, case):
                     walltime = specs[3]
 
                 walltime = self._default_walltime if walltime is None else walltime # last-chance fallback
+            else:
+                # Set the walltime to the correct walltime_format, if set.
+                # Assuming correct format is #H:#M or %H:%M:%S.
+                if walltime_format is not None:
+                    components=walltime.split(":")
+                    if len(components) > len(walltime_format.split(":")):
+                        walltime = components[0] + ":" + components[1]
```

Comment:
You are implying that if a user provides more `:` than the format expects, then the format must expect exactly 2. I think you want

```
walltime = ':'.join(components[:len(walltime_format.split(":"))])
```